ql/patterns/lazyobject.hpp

```diff
@@ -67,22 +68,17 @@ namespace QuantLib {
             method, thus re-enabling recalculations.
         */
         void unfreeze();
-        /*! This method causes the object to forward all
-            notifications, even when not calculated.  The default
-            behavior is to forward the first notification received,
-            and discard the others until recalculated; the rationale
-            is that observers were already notified, and don't need
-            further notification until they recalculate, at which
-            point this object would be recalculated too.  After
-            recalculation, this object would again forward the first
-            notification received.
-
-            \warning Forwarding all notifications will cause a
-                     performance hit, and should be used only when
-                     discarding notifications cause an incorrect
-                     behavior.
+        /*! This method causes the object to forward the first notification received,
+            and discard the others until recalculated; the rationale is that observers
+            were already notified, and don't need further notifications until they
+            recalculate, at which point this object would be recalculated too.
+            After recalculation, this object would again forward the first notification
+            received.
+
+            The behaviour is not always correct though and should only be enabled in
+            appropriate configurations.
         */
-        void alwaysForwardNotifications();
```

Comment:
We can deprecate this one but not remove it right away.  I'd keep it though, in case one calls `forwardFirstNotificationOnly` and wants to switch back.

Reply:
ok - I'll add that back and deprecate it

Reply:
ah now, better keep it, as you say, to revert a change made with `forwardFirstNotificationOnly`